sources/packages/backend/apps/queue-consumers/src/services/student-file/student-file.service.ts

```diff
@@ -27,37 +28,44 @@ export class StudentFileService extends RecordDataModelService<StudentFile> {
   /**
    * Scans the file with the provided unique filename
    * for any viruses.
-   * @param uniqueFileName unique filename of the file to perform the virus scan.
+   * @param jobData virus scan job data.
+   * @param attemptsMade number of attempts made to run this job.
    * @param processSummary process summary logs.
    * @returns boolean true if the file is virus infected, false otherwise.
    */
   async scanFile(
-    uniqueFileName: string,
+    jobData: VirusScanQueueInDTO,
+    attemptsMade: number,
     processSummary: ProcessSummary,
   ): Promise<boolean> {
-    const studentFile = await this.getStudentFile(uniqueFileName);
+    const studentFile = await this.getStudentFile(jobData.uniqueFileName);
     if (!studentFile) {
-      throw new NotFoundException(`Student file ${uniqueFileName} not found.`);
+      throw new CustomNamedError(
+        `Student file ${jobData.uniqueFileName} not found.`,
+        FILE_NOT_FOUND,
+      );
     }
+
     const stream = new Readable();
     stream.push(studentFile.fileContent);
     stream.push(null);
-    const isInfected = await this.clamAVService.scanFile(
-      stream,
-      processSummary,
-    );
-    if (isInfected == null) {
-      // If the file could not be scanned for any reason,
-      // update the file scan status to Pending to keep a
-      // track of the files that haven't been scanned yet.
-      await this.updateFileScanStatus(studentFile.uniqueFileName, isInfected);
+    const virusScanCode = await this.clamAVService.scanFile(stream);
+    if (virusScanCode.errorCode || virusScanCode.isInfected === null) {
+      if (attemptsMade === 11) {
+        await this.updateFileScanStatus(
+          studentFile.uniqueFileName,
+          virusScanCode.isInfected,
+          studentFile.fileName,
+        );
+      }
       throw new CustomNamedError(
-        `Unable to scan the file ${studentFile.uniqueFileName}`,
-        UNABLE_TO_SCAN_FILE,
+        `Unable to scan the file ${studentFile.uniqueFileName} for viruses.`,
+        virusScanCode.errorCode,
       );
     }
+
     let fileName = studentFile.fileName;
-    if (isInfected) {
+    if (virusScanCode.isInfected) {
```

Comment:
Maybe not a part of this PR, but we can remove the else block below because if the `isInfected` is not true does not guarantee us that the file has been scanned and no virus was found. It is possible here that the server was not able to process the request.